fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/RetrieveIndexDAO.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence.jdbc.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.TimeZone;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.ibm.fhir.database.utils.api.IDatabaseTranslator;
+import com.ibm.fhir.persistence.exception.FHIRPersistenceException;
+
+/**
+ * Simple DAO to retrieve index IDs (i.e. logical resource IDs) from the LOGICAL_RESOURCES table.
+ */
+public class RetrieveIndexDAO {
+    private static final Logger logger = Logger.getLogger(RetrieveIndexDAO.class.getName());
+    private static final Calendar UTC_CALENDAR = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+
+    private final IDatabaseTranslator translator;
+    private final String schemaName;
+    private final String resourceTypeName;
+    private final int count;
+    private final Long afterLogicalResourceId;
+    private final Instant notModifiedAfter;
+
+    /**
+     * Public constructor.
+     * @param tx translator
+     * @param schemaName schema name
+     * @param resourceTypeName the resource type of index IDs to return, or null
+     * @param count maximum number of index IDs to return
+     * @param notModifiedAfter only return resources last updated at or before the specified instant, or null
+     * @param afterIndexId only return index IDs after this index ID, or null
+     */
+    public RetrieveIndexDAO(IDatabaseTranslator tx, String schemaName, String resourceTypeName, int count, Instant notModifiedAfter, Long afterIndexId) {
+        this.translator = tx;
+        this.schemaName = schemaName;
+        this.resourceTypeName = resourceTypeName;
+        this.count = count;
+        this.afterLogicalResourceId = afterIndexId;
+        this.notModifiedAfter = notModifiedAfter;
+    }
+
+    /**
+     * Run the DAO command on the database connection.
+     * @param c connection
+     * @return list of logical resource IDs
+     * @throws FHIRPersistenceException
+     */
+    public List<Long> run(Connection c) throws FHIRPersistenceException {
```

Comment:
To avoid any odd behavior by the optimizer, we should use the cached resourceTypeId value and give it as a literal instead of joining against the resource_types table. It will be slightly quicker.

Reply:
Updated.